### PR TITLE
Adds support for replay load form cli argument

### LINKED ISSUE #179: Debug/electron: Read .hsreplay file from args
Would be nice to be able to run `electron . test.hsreplay.xml`

### PR DESCRIPTION
**Pull request by dzannotti**
Closes #179 

`electron . <filename>`
filename can be a relative path too - i have not tried what happens if filename is a url but i strongly suspect it'll fail because of `fs.readFile` usage, would it be useful if this PR should supports that case too?

### CODE REVIEW

File: ts/components/DebugApplication.tsx [hunk 1]
```diff
@@ -48,9 +53,10 @@ export default class DebugApplication extends React.Component<React.ClassAttribu
 
 	public render(): JSX.Element {
 		let widget: JSX.Element = null;
+		const { replay } = this.props
 		switch (this.state.currentWidget) {
 			case Widget.SETUP:
-				widget = <SetupWidget defaultHostname="localhost" defaultPort={9111}
+				widget = <SetupWidget defaultHostname="localhost" defaultPort={9111} autoloadReplay={replay}
```
Comment (dzannotti) [inline ts/components/DebugApplication.tsx]: unsure if this name is right

File: ts/components/SetupWidget.tsx [hunk 2]
```diff
@@ -97,6 +106,18 @@ export default class SetupWidget extends React.Component<SetupWidgetProps, Setup
 		this.loadFile(file);
 	}
 
+	protected preloadReplay(file: string) {
+		this.setState({ working: true });
+		fs.readFile(file, "utf8", (err, data) => {
+  			if (err) {
+  				this.setState({ working: false })
+  				console.error(err)
```
Comment (dzannotti) [inline ts/components/SetupWidget.tsx]: there don't seem to be any error handling in `SetupWidget`, so i just followed the pattern

Comment (beheh) [reply ts/components/SetupWidget.tsx]: Yeah, this is fine.

File: ts/components/DebugApplication.tsx (hunk 1 above)
Comment (beheh) [reply ts/components/DebugApplication.tsx]: Works for me 👍

**Pull request merged**